.github/workflows/rw-preview-label.yml

```diff
@@ -0,0 +1,48 @@
+name: Update Preview Label
+
+# Ensure that only this Pull Request has the preview label.
+
+on:
+  workflow_call:
+
+jobs:
+  preview_labels:
+    name: Update preview labels
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Remove preview label from all other PRs
+      run: |
+        prs=$(curl -L \
+          -H "Accept: application/vnd.github+json" \
+          -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+          -H "X-GitHub-Api-Version: 2022-11-28" \
+          https://api.github.com/search/issues?q=is:pr%20label:preview%20repo:${{ github.repository }} \
+          | jq .items[].number)
+
+        for pr in ${prs}; do
+          if [ "${pr}" -eq "${{ github.event.pull_request.number }}" ]; then
+            continue
+          fi
+
+          curl -L \
+            -X DELETE \
+            -H "Accept: application/vnd.github+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            "https://api.github.com/repos/${{ github.repository }}/issues/${pr}/labels/preview"
+        done
+
+    # When two PRs get the Preview label before they can execute, the first will
+    # remove the label from the second, and the second from the first. This would
+    # leave no PR with a Preview label. So always make sure that if we finish,
+    # our PR has a preview label.
```

Comment:
Doesn't that means both PR will get `preview` label in the end ?

Reply:
Added some extra words to make that a bit more clear :)